### DISCUSSION TITLE
Get modifiers upon keyboard hook event

### DISCUSSION THREAD

**Discussion by alonrks** [TolikPylypchuk/SharpHook#73] [Q&A] [answered]
Hi, I was looking to get via `KeyboardEventData` structure, value indicating the key that was pressed, plus any modifier keys currently pressed. For example, Keys.A|Keys.LControl for LCtrl+A. 
The `KeyboardEventData` class does not provide such information, is there a way to get modifier key state from keyboard state using your library? 
There is a away to get keyboard state via calling GetKeyboardState, handle it in a callback and ensure the keyboard state accurately reflects the modifiers, but I would like to insure with you first, maybe you plan to integrate such important information.

Thank you for your time.

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for asking!

This information is already available, and not just in keyboard events, but in all events. As such, it's contained directly in the `UioHookEvent` instead of `KeyboardEventData` (or `MouseEventData`, or `MouseWheelEventData`). `UioHookEvent` contains the `Mask` field which represents the mask of the currently pressed modifiers. `ModifierMask` is an enum of possible modifiers and has extension methods to query the modifiers in a more convenient way.

You can access the event object inside the global hook event handler through the `RawEvent` property of `HookEventArgs`.
